website/src/developing-extensions/rules.md

```diff
@@ -15,3 +15,11 @@ services:
 ```
 
 For inspiration on how to implement a rule turn to [src/Rules](https://github.com/phpstan/phpstan-src/tree/master/src/Rules) to see a lot of built-in rules.
+
+<div class="bg-yellow-100 border-l-4 border-yellow-500 text-yellow-700 p-4 mb-4" role="alert">
+
+Be aware that PHPStan configuration settings are being cached. You might need to refresh this cache when working on a set of custom rules in order for your phpstan instance to acknowledge your changes.
+
+This cache is stored in the directory path used for temporary files, available through the use of `sys_get_temp_dir()`.
```

Comment:
There's a command for clearing the result cache, please update the text accordingly: https://phpstan.org/user-guide/result-cache#clearing-the-result-cache

Also, this appies to all kinds of extensions so move it to https://phpstan.org/developing-extensions/core-concepts. Thank you!

Reply:
> There's a command for clearing the result cache

Is it 100% the result cache? I also got some usecases where I would add a custom rule, run phpstan, modify the rule, run phpstan again and it would run my previous implementation of the custom rule. My changes in the logic of the rule were not, I believe, acknowledged. I'm not 100% sure as I was messing with lot of things 😅  .

Reply:
Yeah, sounds like result cache.

Reply:
Fixed. I ammended my commit.

I reused the same words as https://phpstan.org/user-guide/result-cache#clearing-the-result-cache as they are cristal clear.

Reply:
Fixed. I ammended my commit.